[java] Add more version shortcuts for older java

hgschmie opened the pull request:
This fixes #1861 

Newer java compiler (java 9+) support the `--release` option to choose
for which version of the JRE to compile. This parameter only takes
simple values as documented by `javac --help`:

```
--release <release>
        Compile for a specific VM version. Supported targets: 6, 7, 8, 9, 10, 11
```

adding these versions to pmd allows the use of the same versions (6, 7
and 8) with PMD as with the compiler, thus removing the need to juggle
between single digit versions for the compiler and 1.<x> for PMD.

<!--
Please, prefix the PR title with the language it applies to within brackets, such as *[java]* or *[apex]*. If not specific to a language, you can use *[core]*
-->

Before submitting a PR, please check that:
 - [x] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `./mvnw clean verify` passes. This will [build](https://github.com/pmd/pmd/blob/master/BUILDING.md) and test PMD, execute PMD and checkstyle rules. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**

pmd-test commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This changeset introduces 0 ne...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/0ad6aec98c494d4a8c4b0938274621f5/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hgschmie commented:
Thank you for the feedback. I went with the most naive approach. I reworked this to ensure that the other versions are now an alias to the existing version. I also added some tests that check the behavior. Let me know if this looks better; I can keep iterating on this to match what is expected.

oowekyala commented on pmd-core/src/main/java/net/sourceforge/pmd/lang/BaseLanguageModule.java:
```diff
@@ -33,6 +33,22 @@ public BaseLanguageModule(String name, String shortName, String terseName, Class
         this.extensions = Arrays.asList(extensions);
     }
 
+    protected void addVersions(LanguageVersionHandler languageVersionHandler, boolean isDefault, String ... languageVersions) {
```
I'm ok with adding this provided you use the `@Experimental` annotation on it. We might want to change that in 7.0, and AFAIK only Java is concerned anyway so the instability is minimal.

hgschmie replied:
done

hgschmie commented:
Done. Too bad this did not make it into the 6.19 release; it would have been nice to have it there.

adangel commented:
> Done. Too bad this did not make it into the 6.19 release; it would have been nice to have it there.

Don't worry. After the release is before the next release...

Maybe we should also update the documentation. For the ant task, we have a list of supported versions (which we unfortunately didn't update with java13 and preview versions):
https://pmd.github.io/latest/pmd_userdocs_tools_ant.html#language-version-selection

We could also improve the doc here: https://pmd.github.io/latest/pmd_userdocs_cli_reference.html - as far as I see, we don't have a example of the version parameter.

I'd also be interested to see, how the CLI behaves, when it is provided with a invalid language version. Do we then display the available versions? And if so, how is this affected by this change? E.g. do we suggest both "1.5" and "5" or just one variant?
